extmod/moduplatform.c

```diff
@@ -95,6 +107,10 @@
 #define MICROPY_HW_MCU_NAME     ""
 #endif
 
+#ifndef MICROPY_HAL_VERSION
+#define MICROPY_HAL_VERSION     ""
+#endif
+
 STATIC const MP_DEFINE_STR_OBJ(info_platform_obj, PLATFORM_SYSTEM "-" MICROPY_VERSION_STRING "-HAL" \
```

Comment:
I think we should remove the `"-HAL"` string here and make ports provide it (they could use SDK or IDF or whatever).  But that's definitely for a separate PR.

Reply:
Yes. I first thought of adding logic to skip the -HAL part when MICROPY_HAL_VERSION is empty but would indeed be better to treat this as 'custom part of platform' like MICROPY_PLATFORM_POSTFIX and have that in mpconfig.h for consistency? Also MICROPY_HW_BOARD_NAME and MICROPY_HW_MCU_NAME are not used, these can be deleted?

Reply:
Yes, those latter two can be removed.

Feel free to open a PR to fix the above items.